### ISSUE TITLE
undesirable_operator_linter doesn't handle `=` as expected

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1191]
I think it's safe to assume that a user specifying `undesirable_operator_linter(list(`=` = NA))` is targeting equals-assignment (even though `assignment_linter()` is more appropriate for this). But HEAD doesn't quite match that:

```
lintr::lint(linters = lintr::undesirable_operator_linter(op = list("=" = NA)), text = "
a = 1
foo <- function(a = 1) { a }
foo(a = 2)
")
```

makes two lints:

```
<text>:2:3: warning: Operator `=` is undesirable.
a = 1
  ^
<text>:4:7: warning: Operator `=` is undesirable.
foo(a = 2)
      ^
```

The latter is a false positive IMO.
